calico_node/tests/st/policy/test_profile.py

```diff
@@ -27,200 +27,277 @@
+        prof_n2['spec']['ingress'].append(rule)
+        _log.info("Profile for network 2: %s", prof_n2)
+
+        rule = {'action': 'allow',
+                'source': {'nets': n2_ips,
+                           'notNets': n2_ips[:1]}}
+        prof_n1['spec']['ingress'].append(rule)
+        self._apply_new_profile(self.new_profiles, self.host1)
+
+        # Check first workload in each group cannot ping the other group.
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n1_workloads[:1],
+                                 fail_list=self.n2_workloads)
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n2_workloads[:1],
+                                 fail_list=self.n1_workloads)
+
+        # Check non-excluded workloads can all ping each other.
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n1_workloads[1:] + self.n2_workloads[1:])
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n2_workloads[1:] + self.n1_workloads[1:])
+
+    def test_rules_dest_ip_nets(self):
+        # Adjust the egress policies to drop all traffic
+        prof_n1, prof_n2 = self._get_profiles(self.new_profiles)
+        prof_n2['spec']['egress'] = []
+        prof_n1['spec']['egress'] = []
+        self._apply_new_profile(self.new_profiles, self.host1)
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n2_workloads[:1],
+                                 fail_list=self.n1_workloads + self.n2_workloads[1:])
+
+        # Add a destination whitelist to n2 that allows pods within it to reach other pods in n2.
+        n2_ips = [str(workload.ip) + "/32" for workload in self.n2_workloads]
+        rule = {'action': 'allow',
+                'destination': {'nets': n2_ips}}
+        prof_n2['spec']['egress'] = [rule]
+        self._apply_new_profile(self.new_profiles, self.host1)
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n2_workloads,
+                                 fail_list=self.n1_workloads)
+
+        # Add some rules that have a single nets entry and multiple notNets entries.  These are
+        # rendered a bit differently in Felix.
+        n1_ips = [str(workload.ip) + "/32" for workload in self.n1_workloads]
+        rule1 = {'action': 'allow',
+                 'destination': {'nets': n1_ips[0:1],
+                                 'notNets': n1_ips[1:]}}
+        rule2 = {'action': 'allow',
+                 'destination': {'nets': n1_ips[1:2],
+                                 'notNets': n1_ips[:1]}}
+        prof_n1['spec']['egress'] = [rule1, rule2]
+        self._apply_new_profile(self.new_profiles, self.host1)
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n1_workloads[:2],
+                                 fail_list=self.n1_workloads[2:])
```

Comment:
Tests we might like to add: 
Rules with: 
- CIDRs other than /32
- the same CIDR twice
- overlapping CIDRs